packages/Sandblocks-Tutorial/SBTutorialStep.class.st

```diff
@@ -236,7 +236,7 @@ SBTutorialStep class >> stepDeleting: anEditor [
 				checkCondition: [:editor | method body statements first submorphs noneSatisfy: [:el | el contents = '6']].
 			step
 				addStep: 'Your current copy buffer is shown in the bottom-right corner. To insert the 6 in the right place, move your cursor using <#moveCursorRight> and <#moveCursorLeft> until you reach the gap just after the 5. A popup will tell you what block the insertion will take place in. Then use <#pasteReplace> to insert the block.'
-				checkCondition: [:editor | method body statements first lastSubmorph contents = '6'].
+				checkCondition: [:editor | method body statements first lastSubmorph submorphBefore contents = '5' and: [method body statements first lastSubmorph contents = '6']].
```

Comment:
By the way, both the old code and the new code would fail if the array was empty or removed. There might also be many other conditions that are not completely fail-safe. Is this an issue, should the `checkCondition` block maybe be wrapped into an `#ifError: [false]` block automatically?

Reply:
That's already done in SBTutorialStep>>#checkConditions :)

Reply:
Aw, perfect! :-) In this case, there already might be some redundant checks (for instance, `args size > 1`), but I guess this is not a problem. :)

Reply:
I used to not have the exception catching because it kept masking actual mistakes I was making. Eventually, I got fed up with all of the guard clauses though and resorted to toggling the exception handler...